tests/conftest.py

```diff
@@ -117,22 +129,20 @@ def global_execution_env():
             # create train data samples
             for i in range(4):
                 spec = f.create_data_sample(datasets=[dataset], test_only=False)
-                data_sample = sess.add_data_sample(spec)
-                assets.train_data_samples.append(data_sample)
+                sess.add_data_sample(spec)
 
             # create test data sample
             spec = f.create_data_sample(datasets=[dataset], test_only=True)
             test_data_sample = sess.add_data_sample(spec)
-            assets.test_data_samples.append(test_data_sample)
 
             # reload datasets (to ensure they are properly linked with the created data samples)
             dataset = sess.get_dataset(dataset.key)
-            assets.datasets.append(dataset)
+            assets._datasets.append(dataset)
```

Comment:
I didn't think this all the way through it seems. I'm not very fond of access private properties directly. Maybe we could have an `add_dataset()` method? It's not the most elegant though. @samlesu any thought on the proper pattern to use here?

Reply:
IMO, it's ok to access private attributes of a class inside its own module. 
It would be indeed better to avoid accessing them outside of the class itself.

To do so, we could use the _TestsAssets constructor to set them inside the `global_execution_env` fixture. Something like:
```python
datasets = []
for ...
    dataset = ...
    datasets.append(dataset)
assets = _TestAssets(datasets=datasets)
```

As the goal is to ensure the tests cannot modify the `_TestAssets` instances, I think it would be a bad idea to implement setter methods.

Reply:
Didn't think of that, this is very clean indeed!